### PR TITLE
Implement get_by_URN_or_create type of function for Contacts

### PR DESCRIPTION
**Pull request by KaitCrawford**

### CODE REVIEW

**Comment by KaitCrawford** [comment]
@rudigiesler ready for review
@rowanseymour This is to retrieve the contact with the given urn and create one if it doesn't exist.

File: casepro/contacts/models.py [hunk 1]
```diff
@@ -184,6 +184,17 @@ def get_or_create(cls, org, uuid, name=None):
             return contact
 
     @classmethod
+    def get_or_create_from_urn(cls, org, path, urn, name=None):
```
Comment (rowanseymour) [inline casepro/contacts/models.py]: Our current nomenclature is urn=scheme+path, but I think it's easiest if we pass around URNs as the complete string, so this becomes just `get_or_create_from_urn(cls, org, urn, name=None)` where `urn` is a string like `tel:+123456`

**Comment by rudigiesler** [comment]
:+1: from me

File: casepro/contacts/models.py [hunk 2]
```diff
@@ -184,6 +184,17 @@ def get_or_create(cls, org, uuid, name=None):
             return contact
 
     @classmethod
+    def get_or_create_from_urn(cls, org, urn, name=None):
+        """
+        Gets an existing contact or creates a stub contact. Used when opening a case without an initial message
+        """
+        contact = cls.objects.filter(urns__contains=[urn]).first()
```
Comment (rowanseymour) [inline casepro/contacts/models.py]: This needs to perform some sort normalization on the URN e.g. tel:+27123456789 should match tel:+27-12345-6789.

If it's helpful you can take a look at the URN class in RapidPro https://github.com/rapidpro/rapidpro/blob/master/temba/contacts/models.py#L64

**Pull request merged**